### PR TITLE
[T4A5][W10-A4] Wang Zexin

### PR DESCRIPTION
**Pull request by wangzexin**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -121,16 +133,51 @@ public void removePerson(ReadOnlyPerson toRemove) throws PersonNotFoundException
...
+    	return tagging;
+    }
 
     /**
      * Clears all persons and tags from the address book.
      */
     public void clear() {
         allPersons.clear();
         allTags.clear();
+        printTaggings(_allTaggings);
```
Comment (hwkchia) [inline src/seedu/addressbook/data/AddressBook.java:161]: Isn't printing taggings supposed to be done just before the program exits?

File: src/seedu/addressbook/data/AddressBook.java [hunk 2]
```diff
@@ -121,16 +133,51 @@ public void removePerson(ReadOnlyPerson toRemove) throws PersonNotFoundException
...
+				out.print("Add ");
+			}
+			else {
+				out.print("Delete ");
+			}
+			out.println(person.toString() + " " + tag.toString());
+		}
+	}
+
+	/**
```
Comment (hwkchia) [inline src/seedu/addressbook/data/AddressBook.java:180]: I guess you need to configure eclipse with the right tab spacing?

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.tag;
+
+import seedu.addressbook.data.person.Person;
+
+public class Tagging{
+	// Types of Tagging
+	public enum TaggingType {ADD, DELETE};
```
Comment (hwkchia) [inline src/seedu/addressbook/data/tag/Tagging.java:7]: 👍

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -22,13 +27,19 @@
 
     private final UniquePersonList allPersons;
     private final UniqueTagList allTags; // can contain tags not attached to any person
+    
+    private static ArrayList<Tagging> _allTaggings;
```
Comment (hwkchia) [inline src/seedu/addressbook/data/AddressBook.java:31]: 👍

**Review by hwkchia — COMMENTED** [review]
Good job with association classes.

**Comment by hwkchia** [comment]
Ack by closing PR

File: src/seedu/addressbook/data/AddressBook.java (hunk 1 above)
Comment (wangzexin) [reply src/seedu/addressbook/data/AddressBook.java:161]: Yes, so this method should be added into the Main.java?

File: src/seedu/addressbook/data/AddressBook.java (hunk 2 above)
Comment (wangzexin) [reply src/seedu/addressbook/data/AddressBook.java:180]: Yes...

**Pull request closed without merging**